doc/install.rst

```diff
@@ -142,6 +147,10 @@ This installs the latest stable release from
 
     pip install pygmt
 
+This upgrades installed PyGMT to be the latest stable release::
+
+    pip install --upgrade pygmt
+
 Alternatively, you can install the latest development version from
```

Comment:
I am still testing how to upgrade the latest dev version.

Reply:
The way I have been doing it is cloning the repository and setting up a second conda environment that doesn't have pygmt installed. I changed into the pygmt folder in the repository, and ran `pip install -e`. The pygmt "version" changes depending on which branch I have checked out. If it's the master branch, it is the latest dev version. It also allows me to change into a different branch to test out a new wrapped function, error message, etc.

Reply:
@willschlitzer yes, your method is similar to the way I do. But the users have to clone the repo themselves.

---

I used `TestPyPI` to install `0.3.1.dev51` several hours ago:
```
pip install --pre --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pygmt
```

Now I ran the command again. the dev pygmt is not updated and is still `0.3.1.dev51`.

---

I also used install `0.3.1.dev51` from PyGMT’s GitHub repository several hours ago:
```
pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt
```
Now I ran the command again, and the pygmt is updated to be `0.3.1.dev52`.
```
Collecting pygmt
  Cloning https://github.com/GenericMappingTools/pygmt.git to /tmp/pip-install-r9g_6l77/pygmt_bb7f2501a6bc48b49048fd305155daf0
  Running command git clone -q https://github.com/GenericMappingTools/pygmt.git /tmp/pip-install-r9g_6l77/pygmt_bb7f2501a6bc48b49048fd305155daf0
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
...
...
Installing collected packages: pygmt
  Attempting uninstall: pygmt
    Found existing installation: pygmt 0.3.1.dev51+g10482f3
    Uninstalling pygmt-0.3.1.dev51+g10482f3:
      Successfully uninstalled pygmt-0.3.1.dev51+g10482f3
Successfully installed pygmt-0.3.1.dev52+g55c6176
```

Therefore, I think we may use `pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt` to install and upgrade dev PyGMT and don't have to clone the repo ourselves.

Reply:
If you use the following command to install the dev version,
```
pip install --pre --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pygmt
```

you can simply add `--upgrade` to update pygmt.

Reply:
> I think we may use `pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt` to install and upgrade dev PyGMT and don't have to clone the repo ourselves.

It's much slower than the `pip install --pre` method and we show both ways in the installing guide.

Reply:
How to upgrade the dev version was added in 2c730e5.